### PR TITLE
[MAINTENANCE] Allowing the tests to run with Docker-in-Windows

### PR DESCRIPTION
**Pull request by Patechoc**
This is a small contribution allowing me to run the tests in a Docker-in-Windows environment.

Changes proposed in this pull request:
- Allowing the DB hostname, **in the tests only**, to be read from the environment variable GE_TEST_LOCAL_DB_HOSTNAME
- Fixing the test of `expect_file_size_to_be_between` where the size of the file was depending on the platform used.

Previous Design Review notes:
- This PR comes after a ZOOM discussion with Eugene and Ben where I presented my issues, one of them being that the tests would not pass on Docker-in-Windows
- and here the reason why I needed a way to replace `localhost` by `host.docker.internal`: https://docs.docker.com/docker-for-windows/networking/#use-cases-and-workarounds

Thank you!
;)
Patrick

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -407,7 +408,8 @@ def get_dataset(
         if not create_engine:
             return None
 
-        engine = create_engine("mysql+pymysql://root@localhost/test_ci")
+        # engine = create_engine("mysql+pymysql://root@localhost/test_ci")
+        engine = create_engine("mysql+pymysql://root@host.docker.internal/test_ci")
```
Comment (spbail) [inline tests/test_utils.py]: Just at first glance while scrolling through this, I believe this should also be `{db_hostname}`? Spotted this in a few other places in this PR too. I haven't looked too deeply yet, so feel free to clarify if I'm mistaken!

Comment (Patechoc) [reply tests/test_utils.py]: Oh Noooo! Sorry for that. You are completely right :/
I will fix that and double-check before marking it.

Comment (Patechoc) [reply tests/test_utils.py]: Hi @spbail,
I did finish to replace my (supposedly temporary) fix with this simple solution using environment variables. Sorry for that again.

I didn't want to impact too much the code, and this is the solution I found to run/pass most of the tests.
I still get some errors though, but when I tried to solve them on my side, they crashed on Azure devops.
I would gladly have a chat with you and your team if you start investigating these few incompatibilities.
At least, for now, I can run the tests dealing with SqlAlchemy! :) 

Cheers!
Patrick

**Review by spbail — APPROVED** [review]
Lgtm, also discussed with @eugmandel - we're good to go!

**Pull request merged**